src/components/views/rooms/NotificationBadge.tsx

```diff
@@ -109,20 +118,22 @@ export default class NotificationBadge extends React.PureComponent<IProps, IStat
     };
 
     public render(): React.ReactElement {
+        const {notification, forceCount, roomId, onClick, ...props} = this.props;
+
         // Don't show a badge if we don't need to
-        if (this.props.notification.color <= NotificationColor.None) return null;
+        if (notification.color <= NotificationColor.None) return null;
 
-        const hasNotif = this.props.notification.color >= NotificationColor.Red;
-        const hasCount = this.props.notification.color >= NotificationColor.Grey;
-        const hasUnread = this.props.notification.color >= NotificationColor.Bold;
+        const hasNotif = notification.color >= NotificationColor.Red;
+        const hasCount = notification.color >= NotificationColor.Grey;
+        const hasUnread = notification.color >= NotificationColor.Bold;
         const couldBeEmpty = (!this.state.showCounts || hasUnread) && !hasNotif;
         let isEmptyBadge = couldBeEmpty && (!this.state.showCounts || !hasCount);
-        if (this.props.forceCount) {
+        if (forceCount) {
```

Comment:
a lot of this will conflict with https://github.com/matrix-org/matrix-react-sdk/pull/4861